pkg/servicelb/controller.go

```diff
@@ -226,21 +227,25 @@ func (h *handler) podIPs(pods []*core.Pod) ([]string, error) {
 
 		for _, addr := range node.Status.Addresses {
 			if addr.Type == core.NodeExternalIP {
-				ips[addr.Address] = true
+				extIPs[addr.Address] = true
+			} else if addr.Type == core.NodeInternalIP {
+				intIPs[addr.Address] = true
 			}
 		}
-		if len(ips) == 0 {
-			for _, addr := range node.Status.Addresses {
-				if addr.Type == core.NodeInternalIP {
-					ips[addr.Address] = true
-				}
-			}
+	}
+
+	keys := func(addrMap map[string]bool) (addrs []string) {
```

Comment:
Small nit: generally want to avoid Hungarian notation in Go.

```suggestion
	keys := func(ips map[string]bool) (addrs []string) {
```